db/migrate/20201126092838_create_user_favourites.rb

```diff
@@ -0,0 +1,15 @@
+class CreateUserFavourites < ActiveRecord::Migration[5.2]
+  def change
+    create_table :user_favourites do |t|
+      t.string    :favourite_type, index: true
+      t.integer   :favourite_id
+      t.integer   :user_id
+      t.boolean   :persistent, default: false 
+
+      t.timestamps
+    end
+
+    add_index :user_favourites, [:favourite_type, :favourite_id], unique: true
```

Comment:
It's good practice to have the reverse index on polymorphic associations as postgresql takes SQL parameter ordering rather seriously. We want best chance of an index match if writing `WHERE record_id=123 and record_type='XZY'`

```suggestion
    add_index :user_favourites, [:favourite_type, :favourite_id], unique: true
    add_index :user_favourites, [:favourite_id, :favourite_type], unique: true
```

Reply:
This is something I learned. Thanks.